### ISSUE TITLE
Start Anaconda Prompt in a Environment where executing Bat-Files from certain locations is restricted

### ISSUE THREAD

**Issue by KoenigMax** [ContinuumIO/anaconda-issues#9450]
### Actual Behavior

When Starting Anaconda Prompt I recieve the Message 

```
Dieses Programm wurde durch eine Gruppenrichtlinie geblockt. Wenden Sie sich an den Systemadministrator um weitere Informationen zu erhalten.
```

This amounts in English to

```
This program was prevented from running by a group policy. Please contact your System Administrator for further Information.
```

After having talked to our System Administration we could gather that the called upon script `%CondaHome%\Scripts\activate.bat` calls a temporary .bat file in the Users Folder. This is not allowed in our System:

```
Die Ausführung von %OSDRIVE%\USERS\E_504KD\APPDATA\LOCAL\TEMP\6\TMP73PTTOPJ.BAT wurde verhindert.
Die Ausführung von %OSDRIVE%\USERS\E_504KD\APPDATA\LOCAL\TEMP\6\TMPVY3DTOM3.BAT wurde verhindert.
```

(Translates to `The execution of ... was prevented.`)

Is there a way to explicitly tell anaconda to use a different temporary location or something like this?

##### Anaconda or Miniconda version:

Anaconda 5.1.0, x86

##### Operating System:

Windows 10

##### `conda info`
I do not know how to run conda without the anaconda prompt running.

**Comment by Kim-Github-2000** [comment]
I have a customer with a similar problem:

Problem is, that per default the Anaconda app (better: the Conda-prompt) is populating the %localappdata%\temp with a unsigned .bat-File that is named randomly everytime (hence getting a new hash-value).

 So all of my applocker possibilities to allow a file (Hash / Path / Signing) are not going to work.

---
Any suggestions?

**Comment by KoenigMax** [comment]
Hello,

in the end our IT department agreed to relax the applocker in this instance. I cannot tell you the exact protocols / rules they enabled.

Best Regards
Maximilian König

**Comment by pnsaevik** [comment]
I have the same issue. There must be a better way for Conda to set environment variables upon activation.

**Comment by nitvit610** [comment]
Currently having this same issue when trying to run conda activate environments through VS Code saying that it's blocked by group policy (AppLocker). Can use conda activate fine within the PowerShell launched through Navigator however. 

We have the Anaconda installation path fully excluded from Applocker, but I think what the previous poster mentioned, there's loads of random bat files in temp that are being created.

Any suggestions?

**Comment by pnsaevik** [comment]
Apart from relaxing the applocker, there is another option, see below. The drawback is that you have to repeat the procedure every time you update conda.

Edit "C:\Program Files\Miniconda3\condabin\_conda_activate.bat" in a suitable text editor.

Go to line 38. Replace the statement @CALL "%_TEMP_SCRIPT_PATH%" with the following 2 lines:

@COPY /Y "%_TEMP_SCRIPT_PATH%" "%SUITABLE_LOCATION%\activate_script.bat" >nul
@CALL "%SUITABLE_LOCATION%\activate_script.bat"

where "%SUITABLE_LOCATION%" is any writeable and executable location. On most machines, C:\Windows\Temp is such a location (see https://pentestlab.blog/2017/05/22/applocker-bypass-weak-path-rules/), but you might as well choose "C:\Program Files\Miniconda3\condabin" if you have write access to that folder.

It is frustrating that a security feature like applocker is very easy to bypass for a malign software, but creates a ton of headache for regular users.

**Comment by Halogen3047** [comment]
Hi @pnsaevik, 

Thanks so much for this workaround. I've actually deployed this to a few of the installations and it's been working a treat.

Unfortunately, it looks like the latest release 2023.03 breaks this fix.
When launching any of the apps, it launches the shortcut bat file from within the user's AppData folder which is blocked by AppLocker.

**Comment by vitoljbw** [comment]
Hey Everyone,

Can confirm the workaround is still functional with the latest miniconda3 py311 installation.

However is there any plans to support customizing this behaviour properly from Anaconda side?
This feels like an extreme hack.

**Comment by kenodegard** [comment]
A fix has been merged into `conda` and will be made available with the `conda 25.3.x` release coming in March, see https://github.com/conda/conda/issues/14632 for the release status

**Comment by jbw-vtl** [comment]
that's great news.
How is this config planned to be exposed?

Tried to look around the release milestones but struggled to find the relevant change.

Within our org, we currently have to maintain an internal patched version of miniconda for essentially this behaviour.
Being able to switch off will be a great win

**Comment by kenodegard** [comment]
My apologies for not including a link to the change itself: https://github.com/conda/conda/pull/14607

This change eliminates `conda activate`'s reliance on executing a dynamically generated script in the `%TEMP%` directory. Instead, `conda activate` will generate a configuration file that a static script (located in `%CONDA_PREFIX%\condabin`) will read to activate environments.

After updating to the latest release, you’ll only need to define an allow rule for the `%CONDA_PREFIX%\condabin` directory.

**Comment by jbw-vtl** [comment]
awesome, sounds good
